src/interrupts/src/lib.rs

```diff
@@ -114,45 +146,62 @@ lazy_static! {
             }; 256]
         };
 
-        idt.set_isr(0, isr0 as u64);
-        idt.set_isr(1, isr1 as u64);
-        idt.set_isr(2, isr2 as u64);
-        idt.set_isr(3, isr3 as u64);
-        idt.set_isr(4, isr4 as u64);
-        idt.set_isr(5, isr5 as u64);
-        idt.set_isr(6, isr6 as u64);
-        idt.set_isr(7, isr7 as u64);
-        idt.set_isr(8, isr8 as u64);
-        idt.set_isr(9, isr9 as u64);
-        idt.set_isr(10, isr10 as u64);
-        idt.set_isr(11, isr11 as u64);
-        idt.set_isr(12, isr12 as u64);
-        idt.set_isr(13, isr13 as u64);
-        idt.set_isr(14, isr14 as u64);
-        idt.set_isr(15, isr15 as u64);
-        idt.set_isr(16, isr16 as u64);
-        idt.set_isr(17, isr17 as u64);
-        idt.set_isr(18, isr18 as u64);
-        idt.set_isr(19, isr19 as u64);
-        idt.set_isr(20, isr20 as u64);
-        idt.set_isr(21, isr21 as u64);
-        idt.set_isr(22, isr22 as u64);
-        idt.set_isr(23, isr23 as u64);
-        idt.set_isr(24, isr24 as u64);
-        idt.set_isr(25, isr25 as u64);
-        idt.set_isr(26, isr26 as u64);
-        idt.set_isr(27, isr27 as u64);
-        idt.set_isr(28, isr28 as u64);
-        idt.set_isr(29, isr29 as u64);
-        idt.set_isr(30, isr30 as u64);
-        idt.set_isr(31, isr31 as u64);
-        idt.set_isr(32, isr32 as u64);
-        idt.set_isr(33, isr33 as u64);
+        idt.set_isr(0, make_idt_entry!(0));
+        idt.set_isr(1, make_idt_entry!(1));
+        idt.set_isr(2, make_idt_entry!(2));
+        idt.set_isr(3, make_idt_entry!(3));
+        idt.set_isr(4, make_idt_entry!(4));
+        idt.set_isr(5, make_idt_entry!(5));
+        idt.set_isr(6, make_idt_entry!(6));
+        idt.set_isr(7, make_idt_entry!(7));
+        idt.set_isr(8, make_idt_entry!(8));
+        idt.set_isr(9, make_idt_entry!(9));
+        idt.set_isr(10, make_idt_entry!(10));
+        idt.set_isr(11, make_idt_entry!(11));
+        idt.set_isr(12, make_idt_entry!(12));
+        idt.set_isr(13, make_idt_entry!(13));
+        idt.set_isr(14, make_idt_entry!(14));
+        idt.set_isr(15, make_idt_entry!(15));
+        idt.set_isr(16, make_idt_entry!(16));
+        idt.set_isr(17, make_idt_entry!(17));
+        idt.set_isr(18, make_idt_entry!(18));
+        idt.set_isr(19, make_idt_entry!(19));
+        idt.set_isr(20, make_idt_entry!(20));
+        idt.set_isr(21, make_idt_entry!(21));
+        idt.set_isr(22, make_idt_entry!(22));
+        idt.set_isr(23, make_idt_entry!(23));
+        idt.set_isr(24, make_idt_entry!(24));
+        idt.set_isr(25, make_idt_entry!(25));
+        idt.set_isr(26, make_idt_entry!(26));
+        idt.set_isr(27, make_idt_entry!(27));
+        idt.set_isr(28, make_idt_entry!(28));
+        idt.set_isr(29, make_idt_entry!(29));
+        idt.set_isr(30, make_idt_entry!(30));
+        idt.set_isr(31, make_idt_entry!(31));
+
+        idt.set_isr(32, make_idt_entry!(isr32, {
+            // timer, do nothing for now
+            pic::eoi_for(32);
```

Comment:
okay so, i added `send_eoi_for` because i was getting compile errors. But now, this line is here and works. Must investigate...